Modify power flow indicator display

ybott-qinetic opened the pull request:
Modify how the animated energy flow is displayed.
When not animated, only show an arrow when power is above 30 in either direction.

Fixes #2395

ybott-qinetic committed:
Modify power flow indicator display

Modify how the animated energy flow is displayed.
When not animated disable the visibility.

Fixes #2395

chriadam commented:
LGTM, but would definitely be good to double check the algorithm, before we merge.  I will ping someone on slack about this.

chriadam commented:
Indeed, thank you for pointing this out!

The _resetDistance() function in WidgetConnector.qml won't re-evaluate if the animation mode changes.  So, better instead to bind the visibility of the Image delegate in electronRepeater (i.e. the electron) to `visible: root.animationMode !== VenusOS.WidgetConnector_AnimationMode_NotAnimated` instead, I think.

chriadam commented:
I also wonder if we need to modify the animationMode associated with other widgetConnectors in the OverviewPage?

e.g. the inverterToAcLoadsConnector -> maybe there should be possible "reverse" flows from AC Loads to the Inverter, in some cases?  If the power can go negative, I guess we should visualise it?

blammit commented on components/widgets/WidgetConnector.qml:
```diff
@@ -60,8 +60,9 @@ Item {
 			? Math.floor(electronTravelDistance / Theme.geometry_overviewPage_connector_electron_interval)
-			: 1 // show just one arrow, if animations are disabled.
-
+			: animationMode !== VenusOS.WidgetConnector_AnimationMode_NotAnimated
```
Could avoid the double negative by changing the condition to check `animationMode === VenusOS.WidgetConnector_AnimationMode_NotAnimated` instead.

chriadam replied:
we shouldn't modify this code, as this method won't re-evaluate when the animation mode changes anyway.  Instead, I suspect that the electron visible property should have a binding on the animationMode.

blammit reviewed (approved):
LGTM, but would like @chriadam to check as well.

chriadam committed:
Set rotation correctly for non-animated electron arrow

When animations are disabled, the path updater's update() function
never runs, and thus the electron arrow's rotation is not
initialised properly.

This commit ensures that in the non-animating case, the rotation
of the electron arrow image is initialised explicitly.

It also resolves an undefined value binding error in mock mode.

chriadam commented:
Found one issue with initialising the electron arrow rotation.  Have added a commit for this - would be good if you could review that one @blammit if possible.

chriadam commented on components/widgets/WidgetConnector.qml:
```diff
@@ -214,6 +214,8 @@ Item {
 					opacity: 0.0
 					source: animationEnabled ? "qrc:/images/electron.svg" : "qrc:/images/electron_arrow.svg"
+					visible: root.animationMode !== VenusOS.WidgetConnector_AnimationMode_NotAnimated
+					rotation: animationEnabled ? undefined : pathUpdater.angleForArrow(pathUpdater.progress, pathUpdater.startToEnd)
```
actually, I can't do this.  undefined is wrong.  I should just set to 0 in that case.  the path updater will update it during animations anyway.

chriadam replied:
Fixed.
